utils/apiCall.ts

```diff
@@ -1,7 +1,9 @@
 // Usefull for the api calls on the backend side, but not only. It is also used as fetcher in hooks
 
+import { getActiveNetworkConfiguration } from '../config/network';
+
 export const apiCall = {
-  baseEndpoint: `${process.env.NEXT_PUBLIC_MULTIVERSX_API}`,
+  baseEndpoint: getActiveNetworkConfiguration().apiAddress,
```

Comment:
@juliancwirko  I noticed this line of code during refactoring. Whats the reason for exposing the baseEndpoint to the consumer? Currently its not used and I'm not sure, why we would ever need that on consumer side. Can I remove this?

Reply:
Yes, I guess it can be just passed where it should be, directly from var.

Also, I would like to overwrite it in `useApiCall`, so by default, it can be Elrond API, but in each function of apiCall, it should allow to overwrite it, but we can leave it for later. I already have it in TODO on the kanban board.

Reply:
oh, and here we will need the public env from the .env file, not the hardcoded Elrond API endpoint. Because it can be any custom API, just public

Reply:
> oh, and here we will need the public env from the .env file, not the hardcoded Elrond API endpoint. Because it can be any custom API, just public

Its not actually hardcoded, you already took care of that: See https://github.com/xdevguild/nextjs-dapp-template/blob/main/config/network.ts#L26

Reply:
Oh wait, nvm, let me think about this for a second.
Do we need another network configuration property?

Reply:
Oh :man_facepalming: ok, so I guess it can stay like that. I'll take a look, and we will merge it. Thanks.

Reply:
I dont think this will work in case of MULTIVERSX_CUSTOM_API set. We have to get it integrated somehow.

Reply:
What do you think about:
apiAddress:
      process.env.NEXT_PUBLIC_MULTIVERSX_API || process.env.MULTIVERSX_CUSTOM_API || process.env.NEXT_PUBLIC_MULTIVERSX_API ||'https://api.elrond.com',

That should do the trick right?

Reply:
I left more comments about it in .env.example and next.config.js

Reply:
So the `NEXT_PUBLIC_MULTIVERSX_API` could always be in use, sometimes it would be a custom **public** API, and sometimes it can define a proxy like `/api/multiversx` when `MULTIVERSX_CUSTOM_API` is set.

We could also change the vars names like for `MULTIVERSX_CUSTOM_API` => `MULTIVERSX_PRIVATE_API`

Reply:
> So the `NEXT_PUBLIC_MULTIVERSX_API` could always be in use, sometimes it would be a custom **public** API, and sometimes it can define a proxy like `/api/multiversx` when `MULTIVERSX_CUSTOM_API` is set.
> 
> We could also change the vars names like for `MULTIVERSX_CUSTOM_API` => `MULTIVERSX_PRIVATE_API`

I think I get what you mean. But is this really the best way of doing it? I mean sometimes you have your custom api in MULTIVERSX_PRIVATE_API and sometimes in NEXT_PUBLIC_MULTIVERSX_API. I dont know, if this will confuse people.

Reply:
On the other hand, I don't have any better ideas, I'll implement it in a second.

Reply:
You are right. A little bit confusing.

Maybe changing the naming convention for the `MULTIVERSX_CUSTOM_API` will clear this out a little bit. It could be `MULTIVERSX_PRIVATE_API`. Then it is clear that it is hidden, and we need a proxy in a different variable, so the same where the public endpoint is also allowed.